qiskit/quantum_info/states/statevector.py

```diff
@@ -677,15 +677,16 @@ def _evolve_instruction(statevec, obj, qargs=None):
                 obj.name, type(obj.definition)))
         if obj.definition.global_phase:
             statevec._data *= np.exp(1j * float(obj.definition.global_phase))
+        qubits = obj.definition.qubits
```

Comment:
`qubits.index(tup)` will repeatedly search through `qubits` to find `tup`, which will likely be slow for large instructions. Can this be a map like
```
qubits = {qubit: i for i, qubit in enumerate(obj.definition.qubits)}
```
? Then `new_qargs` could be something like `[qubits[tup] for tup in qregs]`.

Reply:
Yes, good point. Now i use a map. Also, release note added.